2024-06-20-investigating-code-quality-from-pr-data.md

```diff
@@ -0,0 +1,75 @@
+ "Number of PRs opened per week")
+
+Firstly, a note on PR creation dates. Our tool gathers the 1000 most recent PRs from each repository. For CPython and Rust, these were all created between May and June 2024. In the case of NodeJS and Rust, the earliest PRs come from January and March 2024 respectively. In contrast, .NET, there are PRs from October 2014. Remarkably, there were 257 PRs opened in the CPython in the week commencing 20th May 2024.
+
+##### PR Duration and Review Time
+![png]({{ site.github.url }}/alaws/assets/code-quality/code-quality-analysis-pr-duration.jpg
+ "PR Duration")
+
+`PR Duration` is a measure of the time elapsed between a PR opened, and it ultimately being either merged or closed. However, it may be the case that a PR sits in review for a period of time, before the reviewer first looks at it. Similarly, once it is approved, there may be a time delay before it is merged. For this reason, we added the additional measures of `Time to First Review` and `Time From First Review to Last Commit`. This is perfectly illustrated by .NET, which has a noticeably high PR duration. However, if you raise a PR, you'd spend the majority of this time waiting for the first review, but would be likely to quickly merge/close your PR after the initial review.
+
+##### Contributors
+![png]({{ site.github.url }}/alaws/assets/code-quality/code-quality-analysis-contributions-per-user.jpg
+ "Number of Contributors Per User")
+
+Here, we define a contribution as having made a commit on a PR. The number of commits in that PR aren't considered. All languages seem to accept PRs from new contributors, but .NET and Rust seem to have some very seasoned contributors, with one person having worked on 161 and 157 PRs out of the last 1000.
+
+##### Comment Subject, Tone and Disagreements
+
+![png]({{ site.github.url }}/alaws/assets/code-quality/code-quality-analysis-LLM-pain-points.jpg
+ "LLM Pain Points")
+Here, we asked an LLM to count the number of times reviewers picked up on issues that are characteristics of AI-generated code. It's worth mentioning that all the pain points we identified were picked up to some degree in each of these repositories. Not adhering to project standards is the most frequently identified issue across all of the repositories.
+
+![png]({{ site.github.url }}/alaws/assets/code-quality/code-quality-analysis-comment-tone.jpg
+ "Tone Analysis of Review Comments")
+Here an LLM was asked to take a comment and interpret which of the following tones described the comment: polite, neutral, satisfied, frustrated, excited, impolite, sad. Encouragingly, we see that the reviewers for these repositories are polite in the majority of their comments and are never sad. However, they are frustrated more often than they are excited, especially in the JDK repository.
+ 
+![png]({{ site.github.url }}/alaws/assets/code-quality/code-quality-analysis-number-of-disagreements.jpg
```

Comment:
I feel like this chart could do with a y-axis label. Potentially also changing `MAX` and `AVG` to Maximum and Average respectively may be better.